test/unit-test/cellular_pkthandler_utest.c

```diff
@@ -0,0 +1,978 @@
+    _Cellular_GenericCallback_Ignore();
+    pktStatus = _Cellular_HandlePacket( &context, AT_UNSOLICITED, CELLULAR_SAMPLE_PREFIX_STRING_LARGE_INPUT );
+    TEST_ASSERT_EQUAL( CELLULAR_PKT_STATUS_FAILURE, pktStatus );
+}
+
+/**
+ * @brief Test that NULL parse function AT_UNSOLICITED case for _Cellular_HandlePacket.
+ */
+void test__Cellular_HandlePacket_AT_UNSOLICITED_Null_Parse_Function( void )
+{
+    CellularContext_t context;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    CellularATCommandLine_t testATCmdLine;
+
+    memset( &context, 0, sizeof( CellularContext_t ) );
+    /* copy the token table. */
+    ( void ) memcpy( &context.tokenTable, &tokenTableWoParseFunc, sizeof( CellularTokenTable_t ) );
+    Cellular_ATStrDup_StubWithCallback( _CMOCK_Cellular_ATStrDup_CALLBACK );
+    _Cellular_GenericCallback_Ignore();
+    pktStatus = _Cellular_HandlePacket( &context, AT_UNSOLICITED, CELLULAR_URC_TOKEN_STRING_INPUT );
+    TEST_ASSERT_EQUAL( CELLULAR_PKT_STATUS_FAILURE, pktStatus );
+}
+
+/**
+ * @brief Test that happy path case for _Cellular_HandlePacket.
+ */
+void test__Cellular_HandlePacket_AT_UNSOLICITED_Happy_Path( void )
+{
+    CellularContext_t context;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    CellularATCommandLine_t testATCmdLine;
+
+    memset( &context, 0, sizeof( CellularContext_t ) );
+    /* copy the token table. */
+    ( void ) memcpy( &context.tokenTable, &tokenTable, sizeof( CellularTokenTable_t ) );
+    Cellular_ATStrDup_StubWithCallback( _CMOCK_Cellular_ATStrDup_CALLBACK );
+    _Cellular_GenericCallback_Ignore();
+    pktStatus = _Cellular_HandlePacket( &context, AT_UNSOLICITED, CELLULAR_URC_TOKEN_STRING_INPUT );
+    TEST_ASSERT_EQUAL( CELLULAR_PKT_STATUS_OK, pktStatus );
+
+    /* Test the greater string size in comparison function. */
+    ( void ) memcpy( &context.tokenTable, &tokenTable, sizeof( CellularTokenTable_t ) );
+    Cellular_ATStrDup_StubWithCallback( _CMOCK_Cellular_ATStrDup_CALLBACK );
+    _Cellular_GenericCallback_Ignore();
+    pktStatus = _Cellular_HandlePacket( &context, AT_UNSOLICITED, CELLULAR_URC_TOKEN_STRING_GREATER_INPUT );
+    TEST_ASSERT_EQUAL( CELLULAR_PKT_STATUS_OK, pktStatus );
+
+    /* Test the smaller string size in comparison function. */
+    Cellular_ATStrDup_StubWithCallback( _CMOCK_Cellular_ATStrDup_CALLBACK );
+    _Cellular_GenericCallback_Ignore();
+    pktStatus = _Cellular_HandlePacket( &context, AT_UNSOLICITED, CELLULAR_URC_TOKEN_STRING_SMALLER_INPUT );
+    TEST_ASSERT_EQUAL( CELLULAR_PKT_STATUS_OK, pktStatus );
+}
+
+/**
+ * @brief Test that null buffer AT_UNDEFINED case for _Cellular_HandlePacket.
+ */
+void test__Cellular_HandlePacket_Wrong_RespType( void )
+{
+    CellularContext_t context;
+
+    memset( &context, 0, sizeof( CellularContext_t ) );
+
+    _Cellular_HandlePacket( &context, AT_UNDEFINED, NULL );
+
+    TEST_ASSERT_EQUAL( CELLULAR_PKT_STATUS_OK, queueData );

```

Comment:
Looks odd that the result is OK even when the test says wrong type. Is it intentional?

Reply:
That's the path of the providing parameters, but I think we should return error for this kind of input.
Will fix it.